napari_aicssegmentation/controller/batch_processing_controller.py

```diff
@@ -13,8 +14,56 @@ def __init__(self, application: IApplication, workflow_engine: WorkflowEngine):
         self._workflow_engine = workflow_engine
         self._view = BatchProcessingView(self)
 
+        #Should these go into a model?
+        self.input_folder: Path = None
+        self.output_folder = None
+        self.selected_index = None
+        self.workflow_config = None
+
+
     def index(self):
         self.load_view(self._view)
 
     def run_batch(self):
-        return super().run_batch()
+        workflow = self.get_batch_workflow()
+        workflow.process_all()
+
+    def ready_to_process(self):
+        if not self.workflow_config:
+            return False
+        elif not self.input_folder:
+            return False
+        elif not self.output_folder:
+            return False
+        else:
+            return True
+
+    def select_config(self, selected_config):
+        self.workflow_config = Path(selected_config)
+        if self.ready_to_process():
+            self._view.update_button(enabled=True)
+
+    def select_input_folder(self, input_folder):
+        self.input_folder = Path(input_folder)
+        if self.ready_to_process():
+            self._view.update_button(enabled=True)
+
+    def select_output_folder(self, output_folder):
+        self.output_folder = Path(output_folder)
+        if self.ready_to_process():
+            self._view.update_button(enabled=True)
+
+    def get_batch_workflow(self):
+        if not self.selected_index:
```

Comment:
Don't need to bother with this conditional - the channel index field will default to 0 in the UI, and it also has a default value in the segmenter method signature.